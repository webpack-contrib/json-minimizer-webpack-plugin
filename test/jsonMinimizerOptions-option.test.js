import JsonMinimizerPlugin from "../src/index";

import {
  compile,
  getCompiler,
  getErrors,
  getWarnings,
  readAssets,
} from "./helpers";

describe('when applied with "minimizerOptions" option', () => {
  it('should work "space" options', async () => {
    const testHtmlId = "./simple.json";
    const compiler = getCompiler(testHtmlId);

    new JsonMinimizerPlugin({
      minimizerOptions: {
        space: 2,
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('should work "replacer" options', async () => {
    const testHtmlId = "./simple.json";
    const compiler = getCompiler(testHtmlId);

    new JsonMinimizerPlugin({
      minimizerOptions: {
        replacer: ["squadName", "homeTown"],
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });

  it('should work if "replacer" options is function', async () => {
    const testHtmlId = "./simple.json";
    const compiler = getCompiler(testHtmlId);

    new JsonMinimizerPlugin({
      minimizerOptions: {
        replacer: (key, value) => {
          if (Array.isArray(value)) {
            return undefined;
          }

          return value;
        },
      },
    }).apply(compiler);

    const stats = await compile(compiler);

    expect(readAssets(compiler, stats, /\.json$/i)).toMatchSnapshot("assets");
    expect(getErrors(stats)).toMatchSnapshot("errors");
    expect(getWarnings(stats)).toMatchSnapshot("warnings");
  });
});
