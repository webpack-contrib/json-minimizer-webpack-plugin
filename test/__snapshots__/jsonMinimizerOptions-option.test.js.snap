// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with "minimizerOptions" option should work "replacer" options: assets 1`] = `
{
  "simple.json": "{"squadName":"Super hero squad","homeTown":"Metro City"}",
}
`;

exports[`when applied with "minimizerOptions" option should work "replacer" options: errors 1`] = `[]`;

exports[`when applied with "minimizerOptions" option should work "replacer" options: warnings 1`] = `[]`;

exports[`when applied with "minimizerOptions" option should work "space" options: assets 1`] = `
{
  "simple.json": "{
  "squadName": "Super hero squad",
  "homeTown": "Metro City",
  "formed": 2016,
  "secretBase": "Super tower",
  "active": true,
  "members": [
    {
      "name": "Molecule Man",
      "age": 29,
      "secretIdentity": "Dan Jukes",
      "powers": [
        "Radiation resistance",
        "Turning tiny",
        "Radiation blast"
      ]
    },
    {
      "name": "Madame Uppercut",
      "age": 39,
      "secretIdentity": "Jane Wilson",
      "powers": [
        "Million tonne punch",
        "Damage resistance",
        "Superhuman reflexes"
      ]
    },
    {
      "name": "Eternal Flame",
      "age": 1000000,
      "secretIdentity": "Unknown",
      "powers": [
        "Immortality",
        "Heat Immunity",
        "Inferno",
        "Teleportation",
        "Interdimensional travel"
      ]
    }
  ]
}",
}
`;

exports[`when applied with "minimizerOptions" option should work "space" options: errors 1`] = `[]`;

exports[`when applied with "minimizerOptions" option should work "space" options: warnings 1`] = `[]`;

exports[`when applied with "minimizerOptions" option should work if "replacer" options is function: assets 1`] = `
{
  "simple.json": "{"squadName":"Super hero squad","homeTown":"Metro City","formed":2016,"secretBase":"Super tower","active":true}",
}
`;

exports[`when applied with "minimizerOptions" option should work if "replacer" options is function: errors 1`] = `[]`;

exports[`when applied with "minimizerOptions" option should work if "replacer" options is function: warnings 1`] = `[]`;
